.github/workflows/docs-checks.yml

```diff
@@ -156,7 +156,7 @@ jobs:
       - name: Make Documentation
         working-directory: ./docs/source-${{ matrix.pkg-name }}
         run: |
-          make html --debug --jobs $(nproc) SPHINXOPTS="-W --keep-going"
+          make html --debug --jobs $(nproc) SPHINXOPTS="-W --keep-going" -b linkcheck
```

Comment:
How reliable is this? In my experience, link checkers can be flaky or produce false positives. Let's make sure this does not block code contributions.

Reply:
I updated this based on how its done in [torchmetrics](https://github.com/Lightning-AI/metrics/blob/d2ef383c16f52f02ff38e7c7b8146010e694036b/.github/workflows/docs-check.yml#L79).

Reply:
Let's move this docs build argument to the second PR where all links are fixed...

Reply:
I'll remove it for now then?

Reply:
If it can be done in a separate check that we mark optional, that would be great :)

Reply:
Would it be like another workflow `docs-check-with-linkcheck` but that workflow is optional?

Reply:
I have removed it from `docs-checks.yml` for now.

Reply:
@awaelchli Should I add something like this in the same workflow, not sure if this is how we add an optional check
```
      - name: Make Documentation with Link Check (Optional)
        working-directory: ./docs/source-${{ matrix.pkg-name }}
        run: |
          make html --debug --jobs $(nproc) SPHINXOPTS="-W --keep-going" -b linkcheck
        continue-on-error: true
```

Reply:
Can't really answer these questions. Better to ask @Borda. 
On a high-level, what I'm saying is that we probably want this check to just be informative, but not be a real failure that prevents from merging PRs.

Reply:
yes, this check sometimes produce false positive, especially while checking GH PRs from the changelog, but I would still keep it - just check if we can exclude some files like the changelog from validation...
as for `continue-on-error: true` no as this job is mainly for validation that the docs can build so allowing to continue would not bring anything...

Reply:
so let's keep this PR as it is and create another including adding -b link-check
 OR (better) set link check as a separate step where we allow continue on error
```
$ cd docs
$ make linkcheck
```

Reply:
Resolving this conversation.